src/shared/Paramable.ts

```diff
@@ -88,6 +84,20 @@ export interface ParamInterface {
 
     // Option to hide the description in a tooltip:
     hideDescription?: boolean
+
+    /* An independent validation function, which may validate the value of
+     * this parameter irrespective of the values of other parameters. The
+     * advantage of using this over the aggregate validation function within
+     * `ParamableInterface` is that it runs only when this parameter is updated,
+     * and provides more tailored error messages that appear at the site of
+     * the parameter, rather than above/below the entire list of parameters.
+     * It is also more flexible than the aggregate validation function, working
+     * even when the values of other parameters are currently invalid.
+     * It takes in a realized value of this parameter and returns a validation
+     * status. If this function is not implemented, it assumes that the
+     * parameter must always be valid.
+     */
+    validate?(value: unknown): ValidationStatus
```

Comment:
Ah, I see, this validate now works with strings. Likely it will typically need to call its realize() function; perhaps it should be called with both the string and the realization of that string, of type T? In any case, this can be worked out when there are at least a couple examples of individual-parameter validates() in the code.